assets/js/membraneWebRTC.ts

```diff
@@ -290,6 +312,35 @@ export class MembraneWebRTC {
   public addTrack(track: MediaStreamTrack, stream: MediaStream, trackMetadata: any = {}) {
     this.localTracksWithStreams.push({ track, stream });
     this.localTrackIdToMetadata.set(track.id, trackMetadata);
+    const tracks = this.connection?.getTransceivers().map((trans) => trans.sender.track);
+
+    this.localTracksWithStreams.forEach(({ track, stream }) => {
+      if (!tracks?.includes(track)) this.connection?.addTrack(track, stream);
+    });
+
+    this.connection
+      ?.getTransceivers()
+      .forEach(
+        (trans) => (trans.direction = trans.direction == "sendrecv" ? "sendonly" : trans.direction)
+      );
+
+    if (this.connection) {
```

Comment:
I'd check for this at the beginning of this function and assume connection is defined afterwards